### DISCUSSION TITLE
lc/1086/

### DISCUSSION THREAD

**Discussion by giscus** [doocs/leetcode#2645] [Announcements]
# lc/1086/

多种编程语言实现 LeetCode、《剑指 Offer（第 2 版）》、《程序员面试金典（第 6 版）》题解

https://leetcode.doocs.org/lc/1086/

**Comment by hopeckel** [comment]
public int[][] highFive(int[][] items) {
        if (items == null || items.length == 0) {
            return null;
        }

        Map<Integer, PriorityQueue<Integer>> map = new HashMap<>();
        for (int[] item : items) {
            int id = item[0];
            int score = item[1];
            if (!map.containsKey(id)) {
                map.put(id, new PriorityQueue<>());
            }
            PriorityQueue<Integer> pq = map.get(id);
            pq.offer(score);
            if (pq.size() > 5) {
                pq.poll();
            }
        }
        int[][] res = new int[map.size()][2];
        int index = 0;
        for (Map.Entry<Integer, PriorityQueue<Integer>> entry : map.entrySet()) {
            int id = entry.getKey();
            PriorityQueue<Integer> q = entry.getValue();
            int sum = 0;
            while (!q.isEmpty()) {
                sum += q.poll();
            }
            res[index][0] = id;
            res[index][1] = sum / 5;
            index++;
        }

        return res;
    }
